storage/ndb/include/kernel/Interpreter.hpp

```diff
@@ -85,18 +88,78 @@ class Interpreter {
   static constexpr Uint32 BRANCH_ATTR_OP_PARAM = 26;
   static constexpr Uint32 BRANCH_ATTR_OP_ATTR = 27;
 
+  static constexpr Uint32 LSHIFT_REG_REG = 28;
+  static constexpr Uint32 RSHIFT_REG_REG = 29;
+  static constexpr Uint32 MUL_REG_REG = 30;
+  static constexpr Uint32 DIV_REG_REG = 31;
+  static constexpr Uint32 AND_REG_REG = 32;
+  static constexpr Uint32 OR_REG_REG = 33;
+  static constexpr Uint32 XOR_REG_REG = 34;
+  static constexpr Uint32 NOT_REG_REG = 35;
+  static constexpr Uint32 MOD_REG_REG = 36;
+
+  static constexpr Uint32 ADD_CONST_REG_TO_REG = 37;
+  static constexpr Uint32 SUB_CONST_REG_TO_REG = 38;
+  static constexpr Uint32 LSHIFT_CONST_REG_TO_REG = 39;
+  static constexpr Uint32 RSHIFT_CONST_REG_TO_REG = 40;
+  static constexpr Uint32 MUL_CONST_REG_TO_REG = 41;
+  static constexpr Uint32 DIV_CONST_REG_TO_REG = 42;
+  static constexpr Uint32 AND_CONST_REG_TO_REG = 43;
+  static constexpr Uint32 OR_CONST_REG_TO_REG = 44;
+  static constexpr Uint32 XOR_CONST_REG_TO_REG = 45;
+  static constexpr Uint32 MOD_CONST_REG_TO_REG = 46;
+
+  static constexpr Uint32 READ_PARTIAL_ATTR_TO_MEM = 47;
+  static constexpr Uint32 READ_ATTR_TO_MEM = 48;
+
+  static constexpr Uint32 READ_UINT8_MEM_TO_REG = 49;
+  static constexpr Uint32 READ_UINT16_MEM_TO_REG = 50;
+  static constexpr Uint32 READ_UINT32_MEM_TO_REG = 51;
+  static constexpr Uint32 READ_INT64_MEM_TO_REG = 52;
+  static constexpr Uint32 WRITE_REG_TO_MEM = 53;
+
   /**
    * Macros for creating code
    */
-  static Uint32 Read(Uint32 AttrId, Uint32 Register);
-  static Uint32 Write(Uint32 AttrId, Uint32 Register);
+  static Uint32 Read(Uint32 AttrId, Uint32 RegDest);
+  static Uint32 ReadFull(Uint32 AttrId, Uint32 RegMemOffset, Uint32 RegDest);
```

Comment:
Here on the other hand, I guess it is almost always a constant. I just read the whole column at heap location 35000, to be able to load the whole column.

So I'd always first load-const-u32 the memory offset into a register, and then use the register in the read-full call.

Of course, it works and it gives more flexibility. That said, a variant where the memoffset is a constant is probably more often used.

Reply:
Will take that into consideration,  will have to consider the space of opCodes as well

Reply:
Regarding your problem with ReadFull it is by design.
The first 4 bytes is the header, it contains the read length (lower 16 bits) and the attribute id (higher id bits)
Next comes the 2 bytes little-endian length and finally at pos 6 comes the varbinary.